frontend/ui/lib/widgets/create_trip_widget.dart

```diff
@@ -1,21 +1,39 @@
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 import 'package:tripmeout/widgets/map_widget.dart';
+import 'package:tripmeout/services/trip_service.dart';
+import 'package:tripmeout/model/trip.dart';
+import 'package:tripmeout/model/location.dart';
+import 'package:tripmeout/router/router.dart';
 
 //TODO: Add loading screen after to the Create Trip Widget
 //TODO: Get rid of the newInformation text on the page.
 
 class CreateTripWidget extends StatefulWidget {
+  final TripService tripService;
+
+  CreateTripWidget(this.tripService);
+
   @override
-  _CreateTripWidgetState createState() => _CreateTripWidgetState();
+  _CreateTripWidgetState createState() =>
+      _CreateTripWidgetState(this.tripService);
 }
 
 class _CreateTripWidgetState extends State<CreateTripWidget> {
+  final TripService tripService;
+
+  _CreateTripWidgetState(this.tripService);
+
   String place = "No Input";
   int radius = 0;
   String newInformation = 'Grabbed info placed here.';
-  void changeText() {
+  void submitTrip() {
```

Comment:
Should this method check if `place` was set before doing something?

Reply:
Thank you I didn't think of that. But what I'm going to do now is make the button depend on the state and change the state when the text box has text.